content/docs/grammar-language.md

```diff
@@ -0,0 +1,371 @@
+
+#### Cardinalities
+A cardinality defines the number of elements in a given set. Four different cardinalities can be defined for any expression:
+1. exactly one (no operator)
+2. zero or one (operator `?`)
+3. zero or many (operator `*`)
+4. one or many (operator `+`)
+
+#### Character Range
+The operator `..` is used to declare a character range. It is equivalent to the operator `-` in regular expression. It matches any character in between the left character and the right character (inclusive on both ends).
+```
+terminal INT returns number: ('0'..'9')+;
+```
+is equivalent to the regular expression:
+```
+terminal INT returns number: /[0-9]+/;
+```
+Here, `INT` is matched to one or more characters (by using the operand `+`, which defines a [cardinality](#cardinalities) of 'one or many') between `0` and `9` (inclusive on both ends).
+
+#### Wildcard
+The operator `.` is used to match any character and is similar in regular expression.
+```
+terminal HASHTAG: '#'.+;
+```
+In this example, the terminal rule `HASHTAG` matches a sequence of character starting with `#` followed by one or many ([cardinality](#cardinalities) +) characters.
+
+Equivalent in regular expression:
+```
+terminal HASHTAG: /#.+/;
+```
+#### Until Token
+The operator `->` indicates that all characters should be consumed from the left token *until* the right token occurs. For example, the terminal rule for multi-line comment can be implemented as:
+```
+terminal ML_COMMENT: '/*' -> '*/';
+```
+There is no direct equivalent in regular expression for the until token.
+
+#### Negated Token
+It is possible to negate all tokens using the operator `!`, in regular expression the operator `^` is used for negation.
+```
+terminal BETWEEN_HASHES:'#' (!'#')* '#';
+```
+In regular expression:
+```
+terminal BETWEEN_HASHES: /#[^#]*#/;
+```
+
+#### Rule Calls
+A terminal rule can implement other terminal rules. 
+```
+terminal DOUBLE returns number: INT '.' INT;
+```
+Note: it is easy to create conflict between terminal rules by using *terminal rule calls*. See [Data Type Rules](#data-type-rules) for further details.
+
+#### Alternatives
+It is possible to match several valid options. The terminal rule `STRING` can use alternatives to match a sequence of characters between double quotes `""` or single quotes `''`:
+```
+terminal STRING: /"[^"]*"|'[^']*'/;
+```
+
+#### Groups
+Tokens can be put in sequence specifying the order they have to appear
+```
+terminal FLIGHT_NUMBER: ('A'..'Z')('A'..'Z')('0'..'9')('0'..'9')('0'..'9')('0'...'9')?;
+```
+In this example, the token `FLIGHT_NUMBER` **must** start with two capital letters and followed by three of four digits.
+
+### Terminal Fragments
+Fragments allow for sub-definition of terminal rules to be extracted. They are not consumed by the lexer and have to be consumed by other terminal rules.
```

Comment:
Suggestion: Add here, that normal terminals can also be used as part of other terminals, not only fragments.

Reply:
Isn't that what is documented in the "rule calls" section (101-106) ? Or am I missing something?